lib/Textmaster/Translator/Adapter/AbstractAdapter.php

```diff
@@ -148,10 +172,11 @@ protected function setProperties($subject, array $properties, $language)
      * @param object $subject
      * @param array  $properties Array of 'properties'
      * @param string $language
+     * @param string $activity
      *
      * @return array
      */
-    protected function getProperties($subject, array $properties, $language)
+    protected function getProperties($subject, array $properties, $language, $activity = null)
```

Comment:
You are right !
I now pass it as an argument of function `getPropertyHolder` as property holder could depend on it.